hazelcast-client-legacy/src/test/java/com/hazelcast/client/replicatedmap/ClientReplicatedMapListenerTest.java

```diff
@@ -115,6 +117,50 @@ public void run() throws Exception {
         });
     }
 
+
+    @Test
+    public void testListenToKeyForEntryAdded() throws Exception {
+        HazelcastInstance instance = factory.newHazelcastInstance();
+        HazelcastInstance client = factory.newHazelcastClient();
+        String mapName = randomMapName();
+        ReplicatedMap<Object, Object> replicatedMap = client.getReplicatedMap(mapName);
+        final EventCountingListener listener = new EventCountingListener();
+        replicatedMap.addEntryListener(listener, 1);
+        replicatedMap.put(1, 1);
+        replicatedMap.put(2, 2);
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run() throws Exception {
+                Assert.assertEquals(1, listener.addCount.get());
+            }
+        });
+    }
+
+    @Test
+    public void testListenWithPredicate() throws Exception {
+        HazelcastInstance instance = factory.newHazelcastInstance();
+        HazelcastInstance client = factory.newHazelcastClient();
+        String mapName = randomMapName();
+        ReplicatedMap<Object, Object> replicatedMap = client.getReplicatedMap(mapName);
+        final EventCountingListener listener = new EventCountingListener();
+        replicatedMap.addEntryListener(listener, new FalsePredicate());
+        replicatedMap.put(2, 2);
+        assertTrueEventually(new AssertTask() {
```

Comment:
using assertTrueEventually is probably not the right one. Because in theory it can happen that the assertTrueEventually is run immediately and sees there are not listeners. However the event got pending somewhere and eventually caused the listener to be called due to potential bug. You won't see this because the test thread has already completed the test.

Probably this one is better: assertTrueFiveSeconds